airbyte-integrations/connectors/source-secoda/source_secoda/secoda.yaml

```diff
@@ -0,0 +1,83 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: ["results"]
+  requester:
+    url_base: "https://api.secoda.co"
+    http_method: "GET"
+    authenticator:
+      type: BearerAuthenticator
+      api_token: "{{ config['api_key'] }}"
+  cursor_paginator:
+    type: DefaultPaginator
+    url_base: "*ref(definitions.requester.url_base)"
+    page_token_option:
+      inject_into: path
+    page_size_option:
+      inject_into: body_data
+      field_name: "page_size"
+    pagination_strategy:
+      type: "CursorPagination"
+      cursor_value: "{{ response.links.next }}"
+      stop_condition: "{{ response.links.next is none}}"
+      page_size: 1
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  tables_stream:
+    $ref: "*ref(definitions.base_stream)"
+    retriever:
+      record_selector:
+        $ref: "*ref(definitions.selector)"
+      paginator:
+        $ref: "*ref(definitions.cursor_paginator)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+    $options:
+      name: "tables"
+      primary_key: "id"
+      path: "/table/tables/"
+  terms_stream:
+    $ref: "*ref(definitions.base_stream)"
+    retriever:
+      record_selector:
+        $ref: "*ref(definitions.selector)"
+      paginator:
+        $ref: "*ref(definitions.cursor_paginator)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+    $options:
+      name: "terms"
+      primary_key: "id"
+      path: "/dictionary/terms/"
+  collections_stream:
+    $ref: "*ref(definitions.base_stream)"
+    retriever:
+      record_selector:
+        $ref: "*ref(definitions.selector)"
+      paginator:
+        $ref: "*ref(definitions.cursor_paginator)"
+      requester:
+        $ref: "*ref(definitions.requester)"
```

Comment:
All streams use the same ref, why you don't add those to the base stream and create it using the ref?

Reply:
Indeed.